gilbreth_perception/src/recognition_node.cpp

```diff
@@ -257,19 +272,28 @@ class RecognitionClass {
 
     else {
       std::cerr << "Using Correspondence Grouping" << std::endl;
+      // Extract Scene Keypoint
+      pcl::UniformSampling<PointType> uniform_sampling;
+      pcl::PointCloud<PointType>::Ptr scene_keypoints(new pcl::PointCloud<PointType>());
+      uniform_sampling.setInputCloud(scene);
+      uniform_sampling.setRadiusSearch(key_point_sampling);
+      pcl::PointCloud<int> keypointIndices1;
+      uniform_sampling.compute(keypointIndices1);
+      pcl::copyPointCloud(*scene, keypointIndices1.points, *scene_keypoints);
+      //Compute Descriptor
       pcl::SHOTEstimationOMP<PointType, NormalType, pcl::SHOT352> descr_est;
       descr_est.setRadiusSearch(descr_rad);
       pcl::PointCloud<pcl::SHOT352>::Ptr scene_descriptors(new pcl::PointCloud<pcl::SHOT352>());
-      descr_est.setInputCloud(scene);
+      descr_est.setInputCloud(scene_keypoints);
       descr_est.setInputNormals(scene_normals);
```

Comment:
Are the normals computed from the original cloud or the uniform sampled cloud?

Reply:
This normal is computed based on the downsampled scene (according to the down_sample parameter in the YAML file), but the downsample job is done in the segmentation node, not recognition node.